src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -2498,6 +2498,38 @@ exprt c_typecheck_baset::do_special_functions(
 
     return is_dynamic_object_expr;
   }
+  else if(identifier == CPROVER_PREFIX "LIVE_OBJECT")
+  {
+    if(expr.arguments().size() != 1)
+    {
+      error().source_location = f_op.source_location();
+      error() << "live_object expects one argument" << eom;
+      throw 0;
+    }
+
+    typecheck_function_call_arguments(expr);
+
+    exprt live_object_expr = live_object_exprt(expr.arguments()[0]);
+    live_object_expr.add_source_location() = source_location;
+
+    return live_object_expr;
+  }
+  else if(identifier == CPROVER_PREFIX "WRITEABLE_OBJECT")
```

Comment:
I'm starting to wonder: to what extent is this different from `__CPROVER_w_ok(p, 1)`?

Reply:
Very. The idea is to define `w_ok(p, s) = writeable ∧ live ∧ offset+s≤object_size`.
This won't work in CBMC without major changes, so this will be CHC only for now.